compiler/resolution.common/src/org/jetbrains/kotlin/resolve/calls/results/FlatSignature.kt

```diff
@@ -46,7 +47,8 @@ fun <T> SimpleConstraintSystem.isSignatureNotLessSpecific(
     specificityComparator: TypeSpecificityComparator
 ): Boolean {
     if (specific.hasExtensionReceiver != general.hasExtensionReceiver) return false
-    if (specific.valueParameterTypes.size != general.valueParameterTypes.size) return false
+    if (specific.contextReceiverCount != general.contextReceiverCount) return false
```

Comment:
Not sure if it's correct
returning false in case `specific.contextReceiverCount != general.contextReceiverCount` would mean that `specific` and `general` are incomparable
While I would say we should return true if `specific` has at least one context receiver and `general` has none AND beside that `isSignatureNotLessSpecific` would return true
But this is rather controversial and needs to be redesigned
Please, add this to the list of remaining design questions

Reply:
I agree with the point that we cannot treat them as incomparable. I changed the condition to `specific.contextReceiverCount > general.contextReceiverCount`, which widens your suggestion. So in the case
```kotlin
context(Int, String)
fun f(whatever: Param): Whatever

context(Int)
fun f(whatever: Param): Whatever
```
the first is not less specific than the second and the second is less specific than the first. Seems right, WDYT? 

Added to the list of remaining design questions.